### ISSUE TITLE
ReDoc fails to expand JSON Schema refs in examples

### ISSUE THREAD

**Issue by MateusAzevedo** [Rebilly/ReDoc#840]
**Related to #311 and #380** 

I'm using external schema and example files to reuse them in multple operations. ReDoc correctly render example code for the first operation, but fails to render subsequent ones, rendering something like this:

![captura de tela de 2019-03-08 12-20-27](https://user-images.githubusercontent.com/5148259/54037346-b215fd80-419c-11e9-821b-70479207c251.png)

As @vegyraupe commented in #380: 

> The problem occurs when the same external reference is being used twice. This results in bundle to only include it once and returning a relative reference instead of the dereferenced object.

You asked for a reproducible example, so a created this [Gist](https://gist.github.com/MateusAzevedo/921fae9f4f3369bd44f338e2928eaf6d) to help.

I tested the example spec with "redoc-cli serve", "redoc-cli bundle" and "speccy serve", all result in the same problem.
In the final bundled spec file, the schema and example code are included in the first operation, but the other ones have just a reference to the first operation:

```json
"responses": {
    "200": {
        "description": "OK",
        "content": {
            "application/json": {
                "schema": {
                    "$ref": "#/paths/~1cities/post/responses/201/content/application~1json/schema"
                },
                "example": {
                    "$ref": "#/paths/~1cities/post/responses/201/content/application~1json/example"
                }
            }
        }
    }
}
```

As you can see, both schema and example have a reference to the first operation. ReDoc can render the schema for all operataion (it follows the $ref) but don't do the same for the example.

**Comment by RomanGotsiy** [comment]
Your spec is incorrect according to the OpenAPI 3.0 spec.

`example` field of [Media Type Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#media-type-object) can't be a `$ref` - it is direct example value. (the fact ReDoc dereferences it for the first operation is, in fact, ReDoc bug which I don't think I can fix in short term)

In order to use `$ref` for example values use `examples` field. It supports `$ref` and ReDoc will handle it properly in the upcoming release (I just implemented it).

Here is the updated gist: https://gist.github.com/RomanGotsiy/29152ff5a4da68dbb49131eaaf347530

Closing this issue. Feel free to continue the discussion here.

The new ReDoc version is coming tomorrow.

**Comment by Alexgmin** [comment]
> Your spec is incorrect according to the OpenAPI 3.0 spec.
> 
> `example` field of [Media Type Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#media-type-object) can't be a `$ref` - it is direct example value. (the fact ReDoc dereferences it for the first operation is, in fact, ReDoc bug which I don't think I can fix in short term)

I think you're wrong. According to the spec the type is `any`. Plus there's this in the description:
> Furthermore, if referencing a schema which contains an example, the example value SHALL override the example provided by the schema.

If you couldn't set the `example` to a `$ref` the extra clarification wouldn't be needed. And even if a direct `$ref` wasn't allowed ReDoc doesn't dereference references inside a non direct example.

This works correctly.
```
openapi: '3.0.0'
info:
  title: 'test'
  version: '1.0.0'
paths:
  /hello:
    get:
      description: 'Get welcome message.'
      responses:
        '200':
          description: 'welcome message'
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Message'
              examples:
                default:
                  $ref: '#/components/examples/MessageExample'
components:
  schemas:
    Message:
      type: object
      properties:
        title:
          type: string
        info:
          type: string
  examples:
    MessageExample:
      value:
        foo: 2
        bar: 3
```
While this doesn't
```
openapi: '3.0.0'
info:
  title: 'test'
  version: '1.0.0'
paths:
  /hello:
    get:
      description: 'Get welcome message.'
      responses:
        '200':
          description: 'welcome message'
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Message'
              example:
                foo:
                  $ref: '#/components/examples/MessageExample'
components:
  schemas:
    Message:
      type: object
      properties:
        title:
          type: string
        info:
          type: string
  examples:
    MessageExample:
      value:
        foo: 2
        bar: 3
```

As an addendum, the description of the Media Type Object changed slightly in the 3.0.2 spec, but II don't think it matters here.
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#media-type-object

**Comment by RomanGotsiy** [comment]
It will work using `externalValue` in `examples` once I release a new version. This happens a bit later today.

**Comment by RomanGotsiy** [comment]
`v2.0.0-rc.4` has been just released with the support of `externalValue`

**Comment by Okobiti** [comment]
@RomanHotsiy the gist you provided https://gist.github.com/RomanGotsiy/29152ff5a4da68dbb49131eaaf347530 seems to have been deleted. Any chance you can provide an example of how examples and $ref should work with redoc? I'm using version 2.0.0-rc.18, and having some issues displaying it.

**Comment by jaspersorrio** [comment]
**TL;DR** 

It works if you use `examples` instead of `example`

````
# DOESN'T WORK ×
      example:
        title: Test & Measure
        children:
          - $ref: "#/components/schemas/SubCategory/example"

# WORKS ✓  (Changed "example" -> "examples") 
      examples:
        title: Test & Measure
        children:
          - $ref: "#/components/schemas/SubCategory/example"

````

**Comment by dpkirchner** [comment]
@jaspersorrio Would you mind including that in the context of the original gist? I'm not sure where to add that to get it to render properly in ReDoc.
